bin.js

```diff
@@ -16,17 +20,36 @@ function print (...lines) {
 
 async function run () {
   let arg = process.argv[2] || ''
+  let filter = process.argv[3] || ''
+
   if (arg === '--version') {
     showVersion(print)
-  } else if (arg === '--help') {
+    return
+  }
+
+  if (arg === '--help') {
     showHelp(print)
-  } else if (arg.startsWith('--')) {
+    return
+  }
+
+  if (arg === '--update') {
+    await updateAndShowSnaphots(print, error, cwd, filter)
+    return
+  }
+
+  if (arg === '--watch') {
+    (await watchAndShowSnaphots(print, error, cwd, filter))
+      .on('error', err => error(err.stack || err))
```

Comment:
Let’s refactor this code to something more readable. We even pass `error` function there, we do not need to print errors here.